rules/aws_iam_policy_attachment_has_alternatives.go

```diff
@@ -0,0 +1,74 @@
+package rules
+
+import (
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-aws/project"
+)
+
+// AwsIAMPolicyAttachmentHasAlternativesRule warns that the resource has alternatives recommended
+type AwsIAMPolicyAttachmentHasAlternativesRule struct {
+	tflint.DefaultRule
+
+	resourceType  string
+	attributeName string
+}
+
+// AwsIAMPolicyAttachmentHasAlternativesRule returns new rule with default attributes
+func NewAwsIAMPolicyAttachmentHasAlternativesRule() *AwsIAMPolicyAttachmentHasAlternativesRule {
+	return &AwsIAMPolicyAttachmentHasAlternativesRule{
+		resourceType:  "aws_iam_policy_attachment",
+		attributeName: "name",
+	}
+}
+
+// Name returns the rule name
+func (r *AwsIAMPolicyAttachmentHasAlternativesRule) Name() string {
+	return "aws_iam_policy_attachment_has_alternatives"
```

Comment:
How about a name like `aws_iam_policy_attachment_deprecated`?

Reply:
Or `aws_iam_policy_attachment_exclusive_attachment` may be better.

I prefer names that are descriptive of what issue we are warning about. Additionally, the prefix should preferably match the resource name. What do you think?